### PR TITLE
[stronk] Add new port

### PR DESCRIPTION
**Pull request by anders-wind**
**Describe the pull request**

- #### What does your PR fix?
It adds the stronk library to vcpkg (see https://github.com/twig-energy/stronk)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by anders-wind** [comment]
I don't understand the last error from CI. I could use some pointers.

File: ports/stronk/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
+vcpkg_from_git(
+  OUT_SOURCE_PATH SOURCE_PATH
+  URL https://github.com/twig-energy/stronk
+  REF f298bca3102b48ada516b0b42f2d5e6899cbbc44
+  HEAD_REF main
+)
+
+vcpkg_configure_cmake(
+  SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/stronk/portfile.cmake:9]:
```suggestion
vcpkg_cmake_configure(
  SOURCE_PATH "${SOURCE_PATH}"
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by anders-wind** [comment]
Thanks for the review - I have fixed the issues specified in your comments

File: ports/stronk/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "stronk",
+  "version-semver": "0.2.0",
+  "description": "An easy to customize, strong type library with built in support for unit-like behavior",
+  "homepage": "https://github.com/twig-energy/stronk",
+  "license": "MIT",
+  "builtin-baseline": "acc3bcf76b84ae5041c86ab55fe138ae7b8255c7",
```
Comment (JonLiu1993) [inline ports/stronk/vcpkg.json:7]:
```suggestion
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/stronk/vcpkg.json (hunk 2 above)
Comment (JonLiu1993) [reply ports/stronk/vcpkg.json:7]: We have `ref` in portfile.cmake, why need add `builtin-baseline`?

Comment (anders-wind) [reply ports/stronk/vcpkg.json]: It was taken from the vcpkg.json in our repo, not neccesary here thanks!

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

File: ports/stronk/fix-wrong-include.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+diff --git a/cmake/install-rules.cmake b/cmake/install-rules.cmake
+index 0282aaf..1be8437 100644
+--- a/cmake/install-rules.cmake
++++ b/cmake/install-rules.cmake
+@@ -1,5 +1,5 @@
+ if(PROJECT_IS_TOP_LEVEL)
+-    set(CMAKE_INSTALL_INCLUDEDIR include/stronk CACHE PATH "")
```
Comment (JonLiu1993) [inline ports/stronk/fix-wrong-include.patch:7]: This patch fixes the stronk install .h file in the wrong directory, the old directory is "../include/stronk/stronk/," and the new directory is "../include/stronk/.."

**Comment by JonLiu1993** [comment]
Successfully tested usage in x64-windows triple

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/stronk/fix-wrong-include.patch (hunk 3 above)
Comment (anders-wind) [reply ports/stronk/fix-wrong-include.patch:7]: Thanks for catching this - Ill add the change to the stronk repo for the next release

Comment (anders-wind) [reply ports/stronk/fix-wrong-include.patch:7]: It is merged to main on stronk (`172bffc80e855635bf98c71ebd2f5f2b53995750`) - so updating the ref would render this fix redundant - let me know if you want it to be updated

Comment (JonLiu1993) [reply ports/stronk/fix-wrong-include.patch:7]: We can put this patch here, and when the next version is released, we can update the version to remove this patch

Comment (dg0yt) [reply ports/stronk/fix-wrong-include.patch:7]: This shouldn't be patched at all, but the desired value should be passed as CMake configuration parameter.

File: ports/stronk/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,20 @@
+vcpkg_from_git(
+    OUT_SOURCE_PATH SOURCE_PATH
+    URL https://github.com/twig-energy/stronk
+    REF f298bca3102b48ada516b0b42f2d5e6899cbbc44
+    HEAD_REF main
+    PATCHES
+        fix-wrong-include.patch
```
Comment (dg0yt) [inline ports/stronk/portfile.cmake:7]: Please replace this patch by a CMake configuration parameter.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/stronk/portfile.cmake (hunk 4 above)
Comment (JonLiu1993) [reply ports/stronk/portfile.cmake]: Done

File: ports/stronk/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    URL https://github.com/twig-energy/stronk
+    REF f298bca3102b48ada516b0b42f2d5e6899cbbc44
+    HEAD_REF main
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DCMAKE_INSTALL_INCLUDEDIR="${CURRENT_PACKAGES_DIR}/include"
```
Comment (dg0yt) [inline ports/stronk/portfile.cmake:11]: No.
a) In general, `CMAKE_INSTALL_*DIR` can be used with relative paths.
b) Don't use innner quotes, they are not portable. Quote the whole `-D...` if needed.

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Review by dan-shaw — DISMISSED** [review]

File: ports/stronk/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,20 @@
+vcpkg_from_git(
```
Comment (dan-shaw) [inline ports/stronk/portfile.cmake:1]: We can use `vcpkg_from_github` instead

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**